lib/zwave/ZwaveDevice.js

```diff
@@ -526,25 +526,76 @@ class ZwaveDevice extends MeshDevice {
+	/**
+	 * @param capabilitiesOpts
+	 * @private
+	 */
+	_registerCapabilitiesSet(capabilitiesOpts, fn) {
 
-			const parsedPayload = capabilitySetObj.parser.call(this, value, opts);
-			if (parsedPayload instanceof Error) return Promise.reject(parsedPayload);
+		// Register multiple capabilities with a debouncer
+		this.registerMultipleCapabilityListener(capabilitiesOpts.map(x => x.capability), async (valueObj, optsObj) => {
 
-			try {
-				const commandClass = capabilitySetObj.node.CommandClass[`COMMAND_CLASS_${capabilitySetObj.commandClassId}`];
-				const command = commandClass[capabilitySetObj.commandId];
-				const commandSetPromise = command.call(command, parsedPayload);
-				if (this.node.battery === true && this.node.online === false) return Promise.resolve('TRANSMIT_QUEUED');
-				return commandSetPromise;
-			} catch (err) {
-				return Promise.reject(err);
+			// Let the app try to handle the debounced capabilities updates
+			const result = await fn(valueObj, optsObj);
+
+			// If it did not handle it for some reason, return to the defaults
+			if (!result || result instanceof Error) {
+
+				// Loop all changed capabilities
+				for (const capabilityId of Object.keys(valueObj)) {
+					const capabilityObj = capabilitiesOpts.find(x => x.capability === capabilityId);
+					const commandClassId = capabilityObj.commandClass;
+					const value = valueObj[capabilityId];
+					const opts = optsObj[capabilityId];
+
+					// Try and get capability set object
+					const capabilitySetObj = this._getCapabilityObj('set', capabilityId, commandClassId);
+					if (capabilitySetObj instanceof Error) {
+						this.error(`capabilitySetObj ${capabilityId} ${commandClassId} is error`, capabilitySetObj);
+						break;
+					}
+
+					// Try to handle executing the capability change event
+					try {
+						await this._registerCapabilityListenerHandler(capabilitySetObj, capabilityId, value, opts);
+					} catch (err) {
+						this.error('_registerCapabilityListenerHandler() -> failed', err);
+						break;
+					}
+				}
 			}
-		});
+		}, 500);
+	}
+
+	/**
+	 * @param capabilitySetObj
+	 * @param capabilityId
+	 * @param value
+	 * @param opts
+	 * @returns {Promise.<*>}
+	 * @private
+	 */
+	async _registerCapabilityListenerHandler(capabilitySetObj, capabilityId, value, opts) {
+		if (typeof capabilitySetObj.parser !== 'function') return Promise.reject(new Error('missing_parser'));
+
+		const parsedPayload = capabilitySetObj.parser.call(this, value, opts);
+		if (parsedPayload instanceof Error) return Promise.reject(parsedPayload);
+
+		try {
+			const commandClass = capabilitySetObj.node.CommandClass[`COMMAND_CLASS_${capabilitySetObj.commandClassId}`];
+			const command = commandClass[capabilitySetObj.commandId];
+			const commandSetPromise = command.call(command, parsedPayload);
+			if (this.node.battery === true && this.node.online === false) return Promise.resolve('TRANSMIT_QUEUED');
```

Comment:
Not sure if this is desired, it changes the current behaviour to return directly when it concerns a battery node (opposed to waiting till it is actually set). Does this solve a specific issue for you? Else I would prefer to leave it out.